arangod/Replication2/StateMachines/Document/DocumentStateIndexHandler.h

```diff
@@ -0,0 +1,68 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2023 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Alexandru Petenchea
+////////////////////////////////////////////////////////////////////////////////
+#pragma once
+
+#include "Basics/Result.h"
+#include "Cluster/ClusterTypes.h"
+#include "Replication2/ReplicatedLog/LogCommon.h"
+#include "VocBase/Methods/Indexes.h"
+
+#include <set>
+#include <shared_mutex>
+
+struct TRI_vocbase_t;
+
+namespace arangodb {
+class MaintenanceFeature;
+
+namespace replication2::replicated_state::document {
+struct IDocumentStateIndexHandler {
```

Comment:
I do not See this struct in Action anywhere, it is also not implemented (at least not checked in)
A method with same signature is implemented on the DocumentStateShardHandler.
Was this intentional, is there still a last step to go? Was this file temporary and not fully removed?

Reply:
At first I created this class to handle indexes, but then I realized that indexes are tightly coupled to shards. I moved the functionality to `DocumentStateShardHandler` and removed the .cpp file, but forgot about the header. Thanks for noticing!
Fixed in https://github.com/arangodb/arangodb/pull/19699/commits/101ef1fc6755e2a3bc69dd4f072b76f6be4c98d8